ansible/roles/legend/tasks/xhprof.yml

```diff
@@ -0,0 +1,34 @@
+---
+- name: Set XHprof
+  lineinfile:
+    dest: /etc/php5/fpm/php.ini
+    state: present
+    regexp: '^(;\s*)?\[xhprof\]'
+    line: '[xhprof]'
+- name: Set XHprof parameters
+  lineinfile:
+    insertafter: '^(;\s*)?\[xhprof\]'
+    dest: /etc/php5/fpm/php.ini
+    regexp: "{{ item.regexp }}"
+    line: "{{ item.line }}"
+  with_items:
+    - { regexp: '^(;\s*)?xhprof.output_dir\s*=\s*"/var/tmp/xhprof"', line: 'xhprof.output_dir = "/var/tmp/xhprof"' }
+    - { regexp: '^(;\s*)?extension\s*=\s*xhprof.so', line: 'extension = xhprof.so' }
+  notify: restart php5-fpm
+
+- name: Clone xhgui
+  git:
+    repo: https://github.com/perftools/xhgui.git
+    dest: /home/vagrant/www/localhost/xhgui
+    version: master
+    depth: 1
+    accept_hostkey: True
+- name: Install xhgui
+  sudo: yes
+  sudo_user: root
+  shell: 'cd /home/vagrant/www/localhost/xhgui/ && php install.php'
```

Comment:
The installation should be handled by the provisioning process, no `sudo php`'s in the repo please.
As I can see, it handleds Composer installation(!), does `composer install`, and creates cache directories.